File: DemoServer/AdditionalLanguages/go/Basics/createDocument/CreateDocument.go
package main

//region Usings
import (
    "fmt"
    "github.com/ravendb/ravendb-go-client"
)
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    createDatabase()
    createDocument("aaa","bbb","ccc","ddd")
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

func createDatabase() {
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "testGO"
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    var err = globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        fmt.Printf("d.store.Maintenance().Server().Send(createDatabaseOperation) failed with %s\n", err)
    }
}

//region Demo
func createDocument(companyName, companyPhone, contactName, contactTitle string) error {

    //region Step_1
    newCompany := Company {
        Name:    companyName,
        Phone:   companyPhone,
        Contact: &Contact {
            Name:  contactName,
            Title: contactTitle,
        },
    }
    //endregion

    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()

    //region Step_2
    err = session.Store(&newCompany)
    if err != nil {
        return err
    }
    //endregion

    //region Step_3
    theNewDocumentID := newCompany.ID
    //endregion

    //region Step_4
    err = session.SaveChanges()
    if err != nil {
        return err
    }
    //endregion

    fmt.Printf("Created a new document with id: %s \n", theNewDocumentID)
    return nil   
}

type Company struct {
    ID      string
    Name    string
    Phone   string
    Contact *Contact
}

type Contact struct {
    Name  string
    Title string
}
//endregion

File: DemoServer/AdditionalLanguages/python/basics/createDocument/createDocument.py
from database import DocumentStoreHolder
from models import Company, Contact
from demo_example import Example, RunParamsBase

class RunParams(RunParamsBase):
    def __init__(
        self,
        company_name: str = None,
        company_phone: str = None,
        contact_name: str = None,
        contact_title: str = None,
    ):
        self.company_name = company_name
        self.company_phone = company_phone
        self.contact_name = contact_name
        self.contact_title = contact_title


class CreateDocument(Example):
    def run(self, run_params: RunParams) -> str:
        companyName = run_params.company_name or "Hibernating Rhinos"
        companyPhone = run_params.company_phone or "(+972)52-5486969"
        contactName = run_params.contact_name or "New Contact Name"
        contactTitle = run_params.contact_title or "New Contact Title"

        #region Demo
        #region Step_1
        new_company = Company(
            Name = companyName,
            Phone = companyPhone,
            Contact = Contact(contactName, contactTitle),
        )
        #endregion

        with DocumentStoreHolder.store().open_session() as session:
            #region Step_2
            session.store(new_company)
            #endregion

            #region Step_3
            the_new_document_id = new_company.Id
            #endregion

            #region Step_4
            session.save_changes()
            #endregion
        #endregion

        return the_new_document_id
